### PR TITLE
Implement testing toggle

### PR DESCRIPTION
**Pull request by jjennifergu**
## Overview

Implemented toggle variable in `HomeModelController` to easily switch to dummy data for testing

## Changes Made

- Added dummy eateries in `DummyData`
- Implemented `isTesting` variable in `HomeModelController` that determines whether eateries are retrieved from `DummyData` or backend

## Test Coverage

Playtesting

## Screenshots

<details>

  <summary>Before</summary>

  
  
before 

</details>

<details>

  <summary>After</summary>

after

</details>

### CODE REVIEW

**Review by antoinztte — APPROVED** [review]

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 1]
```diff
@@ -27,7 +29,7 @@ enum DummyData {
         ]),
         MenuCategory(category: "Salads", items: [
             MenuItem(
-                description: "A tortilla-wrapped chicken enchilada with your choice of toppings.",
+                description: "A chopped salad with your choice of toppings.",
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:32]: nice 😋

**Review by ReadeP — APPROVED** [review]

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 2]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+                description: "A mixed salad with your choice of toppings.",
+                healthy: true,
+                name: "Mixed Salad",
+                price: 1100
+            )
+        ])
+    ])
+
+    static let terrace = Eatery(
+        campusArea: "Central",
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:100]: iirc `campusArea` takes on discrete values. You might want to consider turning this into a `static enum`.

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 3]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+        events: (0..<5).map { offset in
+            Event(
+                canonicalDay: Day().advanced(by: offset),
+                endTimestamp: Day().advanced(by: offset).date(hour: 15, minute: 0).timeIntervalSince1970,
+                menu: terraceMenu,
+                startTimestamp: Day().advanced(by: offset).date(hour: 8, minute: 0).timeIntervalSince1970
+            )
+        },
+        id: 1002,
+        imageUrl: nil,
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:110]: Might want to consider adding the image to test how that interacts. Images often are problematic.

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 4]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+    static let roseMenu = Menu(categories: [
+        MenuCategory(category: "Taco Bar", items: [
+            MenuItem(
+                description: "Filling for your taco.",
+                healthy: false,
+                name: "Southwest chicken"
+            ),
+            MenuItem(
+                description: "Filling for your taco.",
+                healthy: false,
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:142]: `true` 😤

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 5]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+        events: (0..<5).map { offset in
+            Event(
+                canonicalDay: Day().advanced(by: offset),
+                endTimestamp: Day().advanced(by: offset).date(hour: 20, minute: 0).timeIntervalSince1970,
+                menu: roseMenu,
+                startTimestamp: Day().advanced(by: offset).date(hour: 7, minute: 0).timeIntervalSince1970
+            )
+        },
+        id: 1003,
+        imageUrl: nil,
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:166]: Same as above.

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 6]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+        campusArea: "West",
+        events: (0..<5).map { offset in
+            Event(
+                canonicalDay: Day().advanced(by: offset),
+                endTimestamp: Day().advanced(by: offset).date(hour: 20, minute: 0).timeIntervalSince1970,
+                menu: roseMenu,
+                startTimestamp: Day().advanced(by: offset).date(hour: 7, minute: 0).timeIntervalSince1970
+            )
+        },
+        id: 1003,
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:165]: Is there a reason why the `id` starts are 1000?

File: Eatery Blue/Eatery Blue/Testing/DummyData.swift [hunk 7]
```diff
@@ -68,5 +70,120 @@ enum DummyData {
...
+        imageUrl: nil,
+        latitude: nil,
+        locationDescription: "West Campus",
+        longitude: nil,
+        menuSummary: "Chef's choice",
+        name: "Rose Dining Hall",
+        onlineOrderUrl: nil,
+        paymentMethods: [.credit, .mealSwipes],
+        waitTimesByDay:  [
+            Day(): WaitTimes(
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/Testing/DummyData.swift:175]: This can be a function since it's the same as above.

**Review by NgaiJustin — APPROVED** [review]
Yeah looks good, left some comments. I noticed you implemented some random sampling. I'm not sure how you intend on using this but if you want it to regenerate you might have to delete/reinstall the app when testing which might be problematic.

**Pull request merged**